wordpress-mobile/WordPress-iOS
# Notifications: readability, autosizing and stackviews

This PR for Notifications brings readability, autosizing changes and layout adaptations, mostly revolving around `UIStackView`.

As a bonus, we also end up touching Reader Comments and Blog Comments views as well, since they're all connected or share views. Particularly, the `ReplyTextView` bar got an overhaul to support better layout via `UIStackView`.

Mostly straightforward changes, not too many surprises here. At some point in the near future I'll refactor the project to share code for tableview estimated heights caching, so there's a bit of code duplication here, in the mean time.

To test:
1. Open Notifications on iPhone 5, 6, 7, Pluses, iPads, iOS 10 and iOS 9.
2. Open each type of notification, badges, comments, likes, posts, follows.
3. Click through any nested levels of notifications as well.
4. Make sure all the views are loading as expected, particularly with readable margins on iPad full-width screens.
5. Also ensure that alignments and cell heights are consistent and as expected.

To test:
1. Open up Reader Comments through Notifications or through Reader posts.
2. Ensure the cells, header and reply text bar layout as expected, with consistent margins.
3. Ensure the reply text bar works as expected.

To test:
1. Open up Comments via My Sites > Blog Details > Comments > Comment.
2. Ensure both the Comments table and Comment detail views layout as expected.
3. Ensure the reply text bar works as expected.

To test:
1. On iPad, be sure to test multiple transitions between multi-tasking sizes for each of the primary VCs touched.
2. Also be sure to attempt state restoration on as many VCs as possible, existing bugs may be lingering here.

- `NotificationsViewController`
- `NotificationDetailsViewControlller`
- `CommentsViewController`
- `CommentViewController`
- `ReaderCommentsViewController`

Needs review: @jleandroperez, @frosty can I request both of you to give this a go? It could use two sets of eyes in varying familiarity here.
